scripts/Test-PendingReboot.ps1

```diff
@@ -0,0 +1,23 @@
+<#
+    Return 0 / SUCCESS if reboot is required!
+#>
+
+$cbsKey = Get-ItemProperty "HKLM:\Software\Microsoft\Windows\CurrentVersion\Component Based Servicing" -ErrorAction Ignore
+if ($cbsKey.PSObject.Properties.name -contains 'RebootPending') {
+    Write-Output "Reboot requried - Component Based Servicing:RebootPending registry value detected"
+    New-Item -Path '/var/run' -Name 'reboot-required' -ItemType File -Force
+}
+
+$autoUpdateKey = Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update" -ErrorAction Ignore
+if ($autoUpdateKey.PSObject.Properties.name -contains 'RebootRequired') {
+    Write-Output "Reboot required - Auto Update:RebootRequired registry value detected"
+    New-Item -Path '/var/run' -Name 'reboot-required' -ItemType File -Force
+}
+
+if (Test-Path -Path '/var/run/reboot-required') {
+    Write-Output "Reboot required"
+    exit 0
```

Comment:
if your intention is to return 0 or 1, you must use `return` keyword and not `exit`. exit kills the PowerShell session where this function runs.

```
PS C:\scripts> $return = .\Test-PendingReboot.ps1 

PS C:\scripts> $return
Reboot not required
```
If you see in this, the return value from the script is not 0 or 1. Instead, it returned Reboot not required. If you replace the Write-Output with Write-Information and exit with return, here is what you will see.

```
PS C:\scripts> $return = .\Test-PendingReboot.ps1 

PS C:\scripts> $return
1
```